controlplane/kubeadm/internal/filters.go

```diff
@@ -163,109 +163,88 @@ func matchesKubeadmConfig(kubeadmConfigs map[string]*bootstrapv1.KubeadmConfig,
 		return "", true, nil
 	}
 
-	// Check if KCP and machine ClusterConfiguration matches, if not return
-	match, diff, err := matchClusterConfiguration(currentKubeadmConfig, kcp)
+	// takes the KubeadmConfigSpec from KCP and applies the transformations required
+	// to allow a comparison with the KubeadmConfig referenced from the machine.
+	desiredKubeadmConfig := getAdjustedKcpConfig(kcp, currentKubeadmConfig)
```

Comment:
What about moving getAdjustedKcpConfig into PrepareKubeadmConfigsForDiff, so we have everything in one func?

Reply:
I would keep it here for now.

We don't need it in the upcoming case where we are going to use PrepareKubeadmConfigsForDiff (right in front of CanUpdateMachine) and I'm going to delete this func entirely as it will be part of ComputeDesiredKubeadmConfig in the next PR